src/Message/SmsMessageResult.php

```diff
@@ -13,147 +13,128 @@
 class SmsMessageResult implements SmsMessageResultInterface {
 
   /**
-   * The status of the message.
+   * The status of the message, or NULL if unknown.
    *
-   * @var bool
+   * @var string|NULL
    */
-  public $status;
+  public $status = NULL;
 
   /**
-   * The translated error message if status is FALSE.
+   * The status message as provided by the gateway API.
    *
    * @var string
    */
-  public $errorMessage;
+  protected $statusMessage = '';
 
   /**
-   * The credits used for this message.
+   * The message delivery reports.
    *
-   * @var integer
+   * @var \Drupal\sms\Message\SmsDeliveryReportInterface[]
    */
-  public $creditsUsed;
+  public $reports = [];
 
   /**
-   * The credit balance after this message is sent.
+   * The credit balance after this message is sent, or NULL if unknown.
    *
-   * @var integer
-   */
-  public $creditBalance;
-
-  /**
-   * The message delivery reports keyed by recipient number.
+   * This number is in the SMS gateway's chosen denomination.
    *
-   * @var \Drupal\sms\Message\SmsDeliveryReportInterface[]
+   * @var float|NULL
    */
-  public $reports;
+  public $creditBalance = NULL;
 
   /**
-   * The messages associated with this result.
+   * The credits consumed to process this message, or NULL if unknown.
    *
-   * @var \Drupal\sms\Message\SmsMessageInterface[]
+   * This number is in the SMS gateway's chosen denomination.
+   *
+   * @var float|NULL
    */
-  public $messages = [];
+  public $creditsUsed = NULL;
 
   /**
-   * Create a new message result based on data supplied in the array.
-   *
-   * @param array $data
-   *   Information to be used to instantiate the SmsMessageResult.
-   */
-  public function __construct($data) {
-    $data += $this->defaultData();
-    $this->status = $data['status'];
-    $this->creditBalance = $data['credit_balance'];
-    $this->creditsUsed = $data['credits_used'];
-    $this->errorMessage = $data['error_message'];
-    $this->reports = $data['reports'];
+   * {@inheritdoc}
+   */
+  public function getStatus() {
+    return $this->status;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getStatus() {
-    return $this->status;
+  public function setStatus($status) {
+    $this->status = $status;
+    return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getReports() {
-    return $this->reports;
+  public function getStatusMessage() {
+    return $this->statusMessage;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getBalance() {
-    return $this->creditBalance;
+  public function setStatusMessage($message) {
+    $this->statusMessage = $message;
+    return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getCreditsUsed() {
-    return $this->creditsUsed;
```

Comment:
Maybe add a separate method `getReportMultiple ()` or something that would return a keyed array. A recipient without a report would have null value.

Anyway, we can have this added when needed since it's  an API addition.